src/Metadata/Property/Factory/SerializerPropertyMetadataFactory.php

```diff
@@ -160,13 +161,15 @@ private function getEffectiveSerializerGroups(array $options, string $resourceCl
             $denormalizationContext = $resourceMetadata->getAttribute('denormalization_context');
         }
 
-        return [$normalizationContext['groups'] ?? null, $denormalizationContext['groups'] ?? null];
+        return [
+            isset($normalizationContext['groups']) ? (array) $normalizationContext['groups'] : null,
```

Comment:
Why having a variable before and not here?

Reply:
Because previously we can reuse the var, here we can't.